lobsterpy/plotting/__init__.py

```diff
@@ -226,3 +236,421 @@ def _broaden(energies: np.ndarray, population: np.ndarray, sigma=None, cutoff=4.
+            )
+        )
+        energy_axis.update(ld.energy_axis_style_dict)
+        cohp_axis = (
+            go.layout.XAxis(title=cohp_label, rangeslider={"visible": rangeslider})
+            if invert_axes
+            else go.layout.YAxis(title=cohp_label)
+        )
+        cohp_axis.update(ld.cohp_axis_style_dict)
+
+        layout = (
+            go.Layout(xaxis=cohp_axis, yaxis=energy_axis)
+            if invert_axes
+            else go.Layout(xaxis=energy_axis, yaxis=cohp_axis)
+        )
+
+        # Create figure object
+        fig = go.Figure(layout=layout)
+        fig.update_layout(ld.layout_dict)
+        fig.update_layout(legend=ld.legend_style_dict)
+
+        # Add all traces to figure
+        for _, val_trace in traces.items():
+            for trace in val_trace:
+                fig.add_trace(trace)
+
+        # Update layout with dropdown menu if label resolved plot
+        if len(traces) > 2:
+            # Update visibility of traces
+            for i, _ in enumerate(fig.data):
+                if i <= len(traces["All"]) - 1:
+                    pass
+                else:
+                    fig.data[i].visible = False
+            # Add dropdown buttons
+            fig.update_layout(
+                updatemenus=[
+                    {
+                        "buttons": [
+                            {
+                                "args": [
+                                    {
+                                        "visible": [
+                                            selected_group == group
+                                            for group, val_trace in traces.items()
+                                            for _trace in val_trace
+                                        ]
+                                    }
+                                ],
+                                "label": selected_group,
+                                "method": "update",
+                            }
+                            for selected_group in traces.keys()
+                        ],
+                        "direction": "down",
+                        "showactive": True,
+                        "active": 0,
+                        "x": 0.5,
+                        "y": 1.15,
+                        "bgcolor": "rgba(255,255,255,0.8)",
+                        "bordercolor": "rgba(0,0,0,0.2)",
+                        "xanchor": "center",
+                        "yanchor": "top",
+                        "font": {"family": "Arial", "color": "#444444", "size": 18},
+                    }
+                ]
+            )
+
+        if xlim:
+            fig.update_xaxes(range=xlim)
+        if ylim:
+            fig.update_yaxes(range=ylim)
+
+        fig.update_yaxes(automargin=True)
+
+        return fig
+
+    @staticmethod
+    def _insert_number_of_bonds_in_label(label, character, number_of_bonds):
+        return label.replace(character, character + " " + number_of_bonds, 1)
```

Comment:
Is there a danger that the label will be e.g. "Na - N" and then calling this with "label, 'N', '3 x'" would give "N 3 xa - N"?

Even though this is a private method it would be helpful to have a docstring with example input and output.

It's also a bit surprising that `number_of_bonds` is a string with an "x" in it.
- this function could receive an int and do the string conversion, rather than the code that calls it
- could we use the unicode multiplication sign × ?

Reply:
- [ ] -  implemented the change suggested as method accepting the int and performing string conversion within the method.
- [ ] - Added docstring with example output

- Adding unicode multiplication is possible but this will deviate from output we get from pymatgen. Wanted to keep it consistent with output from pymatgen https://github.com/materialsproject/pymatgen/blob/1b83245810fd776be9dffac98833c93a9725c957/pymatgen/io/lobster/lobsterenv.py#L564